### PR TITLE
Terminate pod with the exit code of the primary process

### PR DESCRIPTION
**Pull request by vishalbollu**
---

checklist:

- [ ] run `make test` and `make lint`
- [ ] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)
- [ ] update examples
- [ ] update docs and add any new files to `summary.md` (view in [gitbook](https://docs.cortex.dev/v/master) after merging)
- [ ] cherry-pick into release branches if applicable
- [ ] alert the dev team if the dev environment changed

### CODE REVIEW

File: pkg/workloads/cortex/serve/init/bootloader.sh [hunk 1]
```diff
@@ -118,6 +119,37 @@ create_s6_service() {
...
+    # https://wiki.gentoo.org/wiki/S6#Process_supervision
+    # https://skarnet.org/software/s6/s6-svscanctl.html
+    # http://skarnet.org/software/s6/s6-svc.html
+    # http://skarnet.org/software/s6/servicedir.html
+
+    # good pages to read about execline
+    # http://www.troubleshooters.com/linux/execline.htm
+    # https://danyspin97.org/blog/getting-started-with-execline-scripting/
+
+    service_name=$1
```
Comment (RobertLucian) [inline pkg/workloads/cortex/serve/init/bootloader.sh:134]: We can rename this to something like `task_name`.

File: pkg/workloads/cortex/serve/init/bootloader.sh [hunk 2]
```diff
@@ -118,6 +119,37 @@ create_s6_service() {
     chmod +x $dest_script
 }
 
+# terminate pod if this process exits (zero or non-zero exit code)
+create_s6_task() {
+    # good pages to read about s6-overlay
```
Comment (RobertLucian) [inline pkg/workloads/cortex/serve/init/bootloader.sh:124]: Should we move these comments at the top of this script and mention that it's for the `create_s6_service` and `create_s6_task` functions?

File: pkg/workloads/cortex/serve/init/bootloader.sh [hunk 3]
```diff
@@ -140,7 +172,7 @@ if [ "$CORTEX_KIND" = "RealtimeAPI" ]; then
 
 # prepare batch otherwise
 else
-    create_s6_service "batch" "cd /mnt/project && $source_env_file_cmd && exec env PYTHONUNBUFFERED=TRUE env PYTHONPATH=$PYTHONPATH:$CORTEX_PYTHON_PATH /opt/conda/envs/env/bin/python /src/cortex/serve/start/batch.py"
+    create_s6_task "batch" "cd /mnt/project && $source_env_file_cmd && PYTHONUNBUFFERED=TRUE PYTHONPATH=$PYTHONPATH:$CORTEX_PYTHON_PATH /opt/conda/envs/env/bin/python /src/cortex/serve/start/batch.py"
```
Comment (RobertLucian) [inline pkg/workloads/cortex/serve/init/bootloader.sh:175]: I see we removed the `exec` from here. Does this mean we can also take it out from the `nginx` and `uvicorn` services? If I remember correctly, the services need the exec for some reason. And without the exec in, does the pod stop if the task exits with a non-zero exit code?

**Review by RobertLucian — APPROVED** [review]
Just a few comments. Overall LGTM.

File: pkg/workloads/cortex/serve/init/bootloader.sh (hunk 3 above)
Comment (vishalbollu) [reply pkg/workloads/cortex/serve/init/bootloader.sh:165]: Without exec the container does exit with the same exit code as the batch process. The correct exit code shows up in k8s.

**Pull request merged**